website/blog/2024-06-21-AgentEval/index.mdx

```diff
@@ -0,0 +1,182 @@
+    {
+        "name": "Conciseness",
+        "description": "The explanation and method provided should be direct and to the point, avoiding unnecessary steps or complexity.",
+        "accepted_values": ["Very concise", "Concise", "Somewhat verbose", "Verbose"]
+    },
+    {
+        "name": "Relevance",
+        "description": "The content of the response must be relevant to the question posed and should address the specific problem requirements.",
+        "accepted_values": ["Highly relevant", "Relevant", "Somewhat relevant", "Not relevant"]
+    }
+]
+``` 
+
+ 
+
+**Criteria Quantification:** 
+
+During the quantification stage, AgentEval will use the generated criteria (or user defined criteria) to assess a given execution chain to determine how well the application performed.
+
+``` 
+def quantify_criteria( 
+    llm_config: Optional[Union[Dict, Literal[False]]], 
+    criteria: List[Criterion], 
+    task: Task, 
+    test_case: str, 
+    ground_truth: str, 
+) 
+``` 
+
+Parameters: 
+* llm_config (dict or bool): llm inference configuration. 
+* criteria ([Criterion](https://github.com/microsoft/autogen/tree/main/autogen/agentchat/contrib/agent_eval/criterion.py)): A list of criteria for evaluating the utility of a given task. This can either be generated by the `generate_criteria` function or manually created.
+* task ([Task](https://github.com/microsoft/autogen/tree/main/autogen/agentchat/contrib/agent_eval/task.py)): The task to evaluate. It should match the one used during the `generate_criteria` step.
+* test_case (str): The execution chain to assess. 
+* ground_truth (str): The ground truth for the test case. 
+ 
+Example Code: 
+``` 
+quantifier_output = quantify_criteria( 
+    llm_config={"config_list": config_list}, 
+    criteria=criteria, 
+    task=task, 
+    test_case=test_case, 
+    ground_truth="true", 
+) 
+``` 
+
+The output will be a json object consisting of the ground truth and a dictionary mapping each criteria to it's score.  
+
+``` 
+{ 
+  "actual_success": true,
+  "estimated_performance": {
+      "Accuracy": "Correct",
+      "Conciseness": "Concise",
+      "Relevance": "Highly relevant"
+    } 
+} 
+```
+
+## What is next?
+* Enbabling AgentEval in AutoGen Studio for a nocode solution.
+* Fully implementing VerifierAgent in the AgentEval framework.
+
+## Conclusion 
+
+AgentEval represents a significant advancement in the evaluation of LLM-powered applications. By combining the strengths of CriticAgent, QuantifierAgent, and VerifierAgent, the framework offers a robust, scalable, and flexible solution for assessing task utility. This innovative approach not only helps developers understand the current performance of their applications but also provides valuable insights that can drive future improvements. As the field of intelligent agents continues to evolve, frameworks like AgentEval will play a crucial role in ensuring that these applications meet the diverse and dynamic needs of their users. 
```

Comment:
I cannot see the advantage of the "strengths of CriticAgent, QuantifierAgent, and VerifierAgent" from this blog. I know they are discussed in detail in the paper, but this might not be obvious to users who haven't read the paper yet. 

Therefore, I suggest adding a brief sentence that refers to the evaluation/ablation section of the paper.

Reply:
@julianakiseleva, would you have a good snippet to use for this?

Reply:
@BeibinLi indeed we did not really wanted to go into the details of verifier, but at the same time wanted to tell reader we have research results that can verify the work of quartier>

Here is suggested edits on verifier:

### Explanation of VerifierAgent's Role

VerifierAgent ensures the criteria used to evaluate a utility are effective for the end-user, maintaining both robustness and high discriminative power. It does this through two main actions:

1. **Criteria Stability**: 
   - Ensures criteria are essential, non-redundant, and consistently measurable.
   - Iterates over generating and quantifying criteria, eliminating redundancies, and evaluating their stability.
   - Retains only the most robust criteria.

2. **Discriminative Power**: 
   - Tests the system's reliability by introducing adversarial examples (noisy or compromised data).
   - Assesses the system's ability to distinguish these from standard cases.
   - If the system fails, it indicates the need for better criteria to handle varied conditions effectively.

@jluey1 @BeibinLi  do you think it would help?

Reply:
@BeibinLi let's know if this change will address the issue you have raised?

Reply:
I think if we replace the text under the "**VerifierAgent: Ensuring Robustness and Relevance**" section with the bullet points above that would help. @BeibinLi